geth/node/rpc.go

```diff
@@ -94,6 +110,66 @@ func (c *RPCManager) Call(inputJSON string) string {
 	return c.makeJSONErrorResponse(ErrRPCServerTimeout)
 }
 
+// callNodeStream delivers giving request and body content to the external ethereum
+// (infura) RPCServer to process the request and returns response.
+func (c *RPCManager) callUpstreamStream(config *params.NodeConfig, body io.Reader) ([]byte, error) {
+	upstreamURL, err := url.Parse(config.UpstreamConfig.URL)
+	if err != nil {
+		return nil, err
+	}
+
+	httpReq := httptest.NewRequest("POST", "/", body)
+	httpReq.RequestURI = ""
```

Comment:
`http.Server` complains when the giving request object has the underline `http.Request.RequestURI` to be already set because it is done during the processing of the raw request. Hence we need to use `httptest.NewRequest` to allow us set back `RequestURI` to a empty default value.